pkg/clustermesh/mcsapi/service_controller.go

```diff
@@ -78,6 +79,26 @@ func servicePorts(svcImport *mcsapiv1alpha1.ServiceImport) []corev1.ServicePort
 			AppProtocol: port.AppProtocol,
 			Port:        port.Port,
 		})
+		portMap[port.Port] = &ports[len(ports)-1]
+	}
+
+	// Populate the derived service targetPort from the local service so that local
+	// EndpointSlice are generated correctly
+	if localSvc == nil {
+		return ports
+	}
+	for _, port := range localSvc.Spec.Ports {
+		targetPortStr := port.TargetPort.String()
+		if targetPortStr == "" || targetPortStr == "0" {
+			continue
+		}
```

Comment:
Nit. Could be dropped, as that already looks the default value..

Reply:
Which part? It seems to me the default should be int with 0 as a value as is but I am not 100% sure either that if the user enter explicitly an empty string it won't work and be used as a sort of default values until that point which leads me to do check for both here but it may not be necessary though :man_shrugging:

Reply:
Ah sorry, the lines highlighted in the comment. My point is essentially that assigning a default value (int/0) is a no-op, because `TargetPort` is a value, not pointer, and hence already assigned to that value at initialization time. That said, it seems that the API server defaults targetPort to port when unset -- and similarly for (string/"") -- so this is even less of a concern.

Reply:
Ah yep sorry, actually I instinctively tried to filter out the default values, but we don't really need to do this because we are just copying the value over anyway,